File: 29.py
#!/bin/python


def count_powers(limit):

    product = 0
    count = 0
    products = []

    for num in range(2, limit + 1):
        for exp in range(2, limit + 1):
            product = num ** exp

            if product not in products:
                products.append(product)
                count += 1

    return count

if __name__ == '__main__':
    limit = 100
    print "Answer found: ", count_powers(limit)

File: 29.cpp
#include <iostream>
#include <cmath>        // for pow()
#include <vector>       // for vector
#include <algorithm>    // for sort()

using namespace std;


int main() {

    vector < double > products;

    // Gets powers of num**exp, 2 >= num; exp >= 100 and append to products vector
    for (double num = 2; num <= 100; num++) {
        for (double exp = 2; exp <= 100; exp++) {
            products.push_back(pow(num, exp));
        }
    }

    // Resize and remove duplicates
    vector< double >::iterator it;
    sort(products.begin(), products.end());
    it = unique(products.begin(), products.end());
    products.resize(distance(products.begin(), it));

    // Print vector size
    cout << "Answer found: " << products.size() << endl;

    return 0;
}
